routing/routing_tests/routing_algorithm.hpp

```diff
@@ -1,11 +1,67 @@
 #pragma once
 
+#include "routing/base/astar_algorithm.hpp"
 #include "routing/base/routing_result.hpp"
+
 #include "routing/road_graph.hpp"
 #include "routing/router.hpp"
 
+#include <cstdint>
+#include <map>
 #include <string>
 
+namespace routing_test
+{
+using namespace routing;
+
+struct SimpleEdge
+{
+  SimpleEdge(uint32_t to, double weight) : m_to(to), m_weight(weight) {}
+
+  uint32_t GetTarget() const { return m_to; }
+  double GetWeight() const { return m_weight; }
+
+  uint32_t m_to;
+  double m_weight;
+};
+
+class UndirectedGraph : public AStarGraph<uint32_t, SimpleEdge, double>
+{
+public:
+  void AddEdge(Vertex u, Vertex v, Weight w);
+  size_t GetNodesNumber() const;
+
+  // AStarGraph overrides
+  // @{
+  void GetIngoingEdgesList(Vertex const & v, vector<Edge> & adj) override;
+  void GetOutgoingEdgesList(Vertex const & v, vector<Edge> & adj) override;
+  double HeuristicCostEstimate(Vertex const & v, Vertex const & w) override;
+  // @}
+
+private:
+  void GetAdjacencyList(Vertex v, vector<Edge> & adj) const;
+
+  std::map<uint32_t, std::vector<Edge>> m_adjs;
+};
+
+class DirectedGraph
```

Comment:
Why is UndirectedGraph inherited from AStarGraph by DirectedGraph is not?

Reply:
because UndirectedGraph used for routing in tests and DirectedGraph is not